### ISSUE TITLE
Fix build when python3 is default

### ISSUE THREAD

**Issue by VileLasagna** [ROCmSoftwarePlatform/rocBLAS#499]
### What is the expected behavior
You should be able to build the library regardless of whether you have python3 on your system

### What actually happens
The scripts just do a very simplistic "well, if /usr/bin/python exists, use that" but that is very likely
to be a symlink to either python3 or python2. If it IS python3, then the build fails

### How to reproduce
Have your `/usr/bin/python` be a symlink pointing to python3 instead instead of python2

### Environment
Observed in Arch Linux which will have python3 as the default.

#### Fix suggestion and discussion

I initially did a survey of three possible options but upon closer look turns out the problem was simpler than I initially surmised so the simplest one turned out to be sufficient. I'm making the fork and the PR

> **TL;DR:** I want to have this fixed and don't mind doing it. But need to know what solution you prefer

~I came to this issue through a current effort to adapt the scripts in [Experimental ROC](https://github.com/RadeonOpenCompute/Experimental_ROC) to Arch Linux, since I didn't seem to have a lot of luck with the AUR packages. I'm picking up small details here and there in the build scripts of the individual packages and am willing to help, especially when, such as in situations like these, it's an easy-looking tweak.
The problem seems to be located in `cmake/virtualenv.cmake`. There actually IS a TODO comment related to this and a commented out call to the cmake module `findPythonInterp`.
This seems to be unnecessary as a quick fix is to simply wrap the `find_program` call in the other condition:
```cmake
find_program(VIRTUALENV_PYTHON_EXE python2)
if(NOT VIRTUALENV_PYTHON_EXE)
    find_program(VIRTUALENV_PYTHON_EXE python)
endif()
```
